### DISCUSSION TITLE
Proposition

### DISCUSSION THREAD

**Discussion by olarno** [Mysterious-Developers/InfinIdea#51] [Ideas]
Bonjour, 

afin de permettre une contribution plus simple, serait-il possible de mettre en place une solution permettant de mettre la solution en local afin de pouvoir réaliser des POC, ajout de fonctionnalités ou même de simple bug fixe. 

😉 

Peut être ajouter dans le readme quelques informations : stack technique, framework ou non, etc... ? 

C'est une simple idée mais qui faciliterais les contributions je pense

**Comment by pierrbt** [comment]
C'est une bonne idée !
Tu veux dire une sorte de version test sur laquelle chacun peut réaliser ses tests ?
Le problème c'est donc qu'il faudrait que ça créé une base de données en local avec des valeurs dedans.
C'est pas impossible à faire avec un script mais donc c'est quand même un peu compliqué surtout si il faut installer un moteur PHP, une db et tout ça ...
Peut être faire une version dev sur une adresse annexe et reliée à une branche sur laquelle tout le monde peut push librement (en fait c'est une très mauvaise idée je viens de réfléchir) ?
Je ne sais pas trop si c'est réalisable.
A toi de me dire ce que tu penserais le plus simple

---

Et oui il faudrait que je rajoutes quelques informations sur le README mais pour faire simple : 
PHP 8.1 qui sert pour générer les pages ainsi que faire le lien avec le JS client avec une sorte d'API ( souvent 1 fonction par fichier PHP comme pour publier un article, se login, etc)
ES6 client-side pour pouvoir envoyer les données
MariaDB pour la base de données reliée à l'API
Pas de framework mais peut être plus tard (le dev front (@NicolasFasa) est pour l'instant en native css/js mais il va apprendre React je crois)

Dis-moi si tu as des suggestions ou d'autres questions.

Pierre
**PS:** J'ai ouvert #52 et #53 pour tes propositions, et je t'avais répondu dans #44

**Reply by olarno** [reply]
> Peut être faire une version dev sur une adresse annexe et reliée à une branche sur laquelle tout le monde peut push librement (en fait c'est une très mauvaise idée je viens de réfléchir) ?

L'idée global (Open source et autre) une branche (main/master/production) le nom que tu veux, quand un contributeur veux pousser une modification (bug fix - feature - etc..) il faut créer une branche liée à une issue. 
Le contributeur créer une branche pousse sa modification et pousse sa branche sur le repo, il fera du coup une Pull Request.
La Pull Request, tu peux ajouter un template afin de "forcer" la standardisation des PR.
Il y a même la possibilité de rajouter des checks diverses plus lancement de test etc... 
Cette PR doit être review par un "contributeur" et un "mainteneur". 
Seul les plus au cœur du projet doivent pourvoir merger les PR dans la branche main.

**Reply by pierrbt** [reply]
Oui je sais je m'étais juste mal exprimé, et oui je vais faire un modèle pour qu'ils expliquent ce qu'ils ont fait 👍

**Comment by pierrbt** [comment]
Je ne pense pas que ce projet va encore beaucoup avancer et cette contribution n'avance plus donc je vais la fermer.
